src/github.com/t3chguy/riot-static/mxclient/room-state.go

```diff
@@ -1,198 +0,0 @@
-		}
-
-		if usePrevContent {
-			if membership, ok := event.PrevContent["membership"].(string); ok {
-				currentMemberState.Membership = membership
-			}
-			if avatarUrl, ok := event.PrevContent["avatar_url"].(string); ok {
-				currentMemberState.AvatarURL = *NewMXCURL(avatarUrl, rs.client.HomeserverURL.String())
-			}
-			if displayName, ok := event.PrevContent["displayname"].(string); ok {
-				currentMemberState.DisplayName = displayName
-			}
-		}
-
-		if membership, ok := event.Content["membership"].(string); ok {
-			currentMemberState.Membership = membership
-		}
-		if avatarUrl, ok := event.Content["avatar_url"].(string); ok {
-			currentMemberState.AvatarURL = *NewMXCURL(avatarUrl, rs.client.HomeserverURL.String())
-		}
-		if displayName, ok := event.Content["displayname"].(string); ok {
-			currentMemberState.DisplayName = displayName
-		}
-
-		rs.memberList = rs.CalculateMemberList()
-	case "m.room.power_levels":
-		// ez convert to powerLevels
-		if data, err := json.Marshal(event.Content); err == nil {
-			var powerLevels PowerLevels
-			err = json.Unmarshal(data, &powerLevels)
-			if err == nil {
-				rs.PowerLevels = powerLevels
-			}
-		}
-
-	case "m.room.name":
-		if name, ok := event.Content["name"].(string); ok {
-			rs.Name = name
-		}
-	case "m.room.topic":
-		if topic, ok := event.Content["topic"].(string); ok {
-			rs.Topic = topic
-		}
-	case "m.room.avatar":
-		if url, ok := event.Content["url"].(string); ok {
-			rs.AvatarURL = *NewMXCURL(url, rs.client.HomeserverURL.String())
-		}
-	}
-}
-
-func (rs *RoomState) CalculateMemberList() []*MemberInfo {
-	memberList := make([]*MemberInfo, 0, len(rs.MemberMap))
-	for _, member := range rs.MemberMap {
-		if member.Membership == "join" {
-			memberList = append(memberList, member)
-		}
-	}
-
-	return memberList
-}
-
-func (rs RoomState) Members() []*MemberInfo {
-	return rs.memberList
-}
-
-func (rs RoomState) Servers() StringIntPairList {
-	serverMap := make(map[string]int)
-	for _, member := range rs.CalculateMemberList() {
-		if mxidSplit := strings.SplitN(member.MXID, ":", 2); len(mxidSplit) == 2 {
-			serverMap[mxidSplit[1]]++
-		}
-	}
-
-	serverList := make(StringIntPairList, 0, len(serverMap))
-	for server, num := range serverMap {
-		serverList = append(serverList, StringIntPair{server, num})
-	}
-
-	sort.Sort(sort.Reverse(serverList))
-	return serverList
-}
-
-// Partial implementation of http://matrix.org/docs/spec/client_server/r0.2.0.html#calculating-the-display-name-for-a-room
-// Does not handle based on members if there is no Name/Alias (yet)
-func (rs RoomState) CalculateName() string {
-	if rs.Name != "" {
-		return rs.Name
-	}
-	if rs.canonicalAlias != "" {
-		return rs.canonicalAlias
-	}
-	if len(rs.Aliases) > 0 {
-		return rs.Aliases[0]
-	}
-
-	return "Empty Room"
-}
-
-func (rs RoomState) CanonicalAlias() string {
-	if rs.canonicalAlias != "" {
-		return rs.canonicalAlias
-	}
-	if len(rs.Aliases) > 0 {
-		return rs.Aliases[0]
```

Comment:
You probably shouldn't use an arbitrary alias as the canonical alias if there is no canonical alias. We used to do that in Riot and it led to some rooms having rather peculiar names..